frb_codegen/src/generator/rust/ty.rs

```diff
@@ -1,9 +1,17 @@
+use std::borrow::Cow;
+
 use crate::generator::rust::*;
+use crate::target::Acc;
 use enum_dispatch::enum_dispatch;
 
 #[enum_dispatch]
 pub trait TypeRustGeneratorTrait {
-    fn wire2api_body(&self) -> Option<String>;
+    fn wire2api_body(&self) -> Acc<Option<String>>;
+
+    /// Handles JsValue to Self conversions.
+    fn wasm2api_body(&self) -> Option<Cow<str>> {
```

Comment:
For WASM, the distinction comes from the fact that `wire2api` handles canonical counterparts while `wasm2api` handles JsValues. There are two ways to remove this:

1. Emit "WASM wire structs" with getters for attributes, replacing arrays as containers. This adds some complexity both to Dart and Rust, but this means we get to work solely with canonical types. (This is also the approach I took in the first pass of this PR.)

2. Handle conversions only from JsValues. However, we would be reinventing the wheel here by opting out of the work that would normally be wasm_bindgen's responsibility.

Also, `wasm2api` sometimes delegates to `wire2api` as well to complete the `JsValue -> canonical wire -> API` flow.

Reply:
I see. So is it possible to have something like wire2api_common, wire2api_io, wire2api_wasm, just like the Acc does?

Reply:
I don't think this is needed yet, since only WASM suffers from this particular paper cut.

Reply:
Then maybe at least we should say the "wasm" in wasm2api is indeed "wire_wasm", i.,e. `wire_wasm_2_api_body`? Since when reading that, I get confused what indeed wasm means (wire wasm? api wasm? something else?)

P.S. If the modification is more than renaming a method, throw it to backlog :)

Reply:
It's not perfect, but how about `wire2api_jsvalue`?

Reply:
Seems ok